### PR TITLE
Added flags for using local images in sandbox

### PR DESCRIPTION
**Pull request by evalsocket**
# TL;DR
Currently, we always pull images from the registry but there are some cases where we want to use local images. 

Small use case -
In flyte repo I want to test sandbox images before releasing them, For every pr, we don't push images, Those images are only available in the local registry. 

Test:
```
~/workspace/flyteorg/flytectl on feature/use-local-sandbox-images *26 +3 !4 ?13 > go run main.go sandbox start --image test --imagePullPolicy Always     
🧑‍🏭 Bootstrapping a brand new flyte cluster... 🔨 🔧
delete existing sandbox cluster [y/n]: y
🐋 pulling docker image for release test
🧑‍🏭 booting Flyte-sandbox container
92021-11-12T09:59:59.047262600Z Starting Docker daemon...
%2021-11-12T10:00:08.046167300Z Done.
72021-11-12T10:00:08.046316000Z Starting k3s cluster...
%2021-11-12T10:00:20.083077100Z Done.
^Csignal: interrupt


~/w/flyteorg/flytectl on feature/use-local-sandbox-images *26 +3 !4 ?13 > go run main.go sandbox start --image test   --imagePullPolicy Never                                                                          
🧑‍🏭 Bootstrapping a brand new flyte cluster... 🔨 🔧
delete existing sandbox cluster [y/n]: y
🐋 pulling docker image for release test
Error: Error response from daemon: pull access denied for test, repository does not exist or may require 'docker login': denied: requested access to the resource is denied
exit status 1

```

## Type
- [ ] Bug Fix
- [x] Feature
- [ ] Plugin

## Are all requirements met?

- [x] Code completed
- [x] Smoke tested
- [x] Unit tests added
- [ ] Code documentation added
- [ ] Any pending items have an associated Issue

## Complete description
_How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/flyteorg/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/flyteorg/flyte/issues/<number>_

### CODE REVIEW

**Comment by pmahindrakar-oss** [comment]
Local is still a registry .
If we are trying to use cached image then should that be driven through a docker image pull policy instead of this explicit local flag

File: cmd/config/subcommand/sandbox/sandbox_config.go [hunk 1]
```diff
@@ -18,4 +20,8 @@ type Config struct {
 	// Flyte compliant sandbox image. Usually useful, if you want to push the image to your own registry and relaunch
 	// from there.
 	Image string `json:"image" pflag:",Optional. Provide a fully qualified path to a Flyte compliant docker image."`
+
+	// Optionally it is possible to use local sandbox image
+	// If local flag pass then flytectl will not pull image from registry. Usually useful, if you want to test your local images without pushing them to a registry
+	Local bool `json:"local" pflag:",Optional. Enable if you want to use local available image."`
```
Comment (EngHabu) [inline cmd/config/subcommand/sandbox/sandbox_config.go:26]: I would rather have the option be called ``ImagePullPolicy`` since this may be needed regardless of whether the image is built locally or not... it could be due to auth, somebody will pull the image separately with special authentication, then call ``flytectl sandbox start --imagePullPolicy Never`` to just run it...

```suggestion
	ImagePullPolicy ImagePullPolicy `json:"imagePullPolicy" pflag:",Optional.Defines the image pull behavior."`
```

with
```golang
//go:generate enumer -type=ImagePullPolicy -trimprefix=ImagePullPolicy

type ImagePullPolicy int
const(
    ImagePullPolicyAlways ImagePullPolicy = iota
    ImagePullPolicyIfNotPresent
    ImagePullPolicyNever
)
```

Comment (evalsocket) [reply cmd/config/subcommand/sandbox/sandbox_config.go:26]: @EngHabu enumer works fine but pflag is not working with enumer. Got this

```
INFO[0000] [image] is of a basic type with default value [*new(string)]. 
Error: invalid type. it must be struct, received [int] for field [imagePullPolicy]
Usage:
  pflags MyStructName --package myproject/mypackage [flags]
  pflags [command]

Examples:

// go:generate pflags MyStruct
type MyStruct struct {
	BoolValue        bool              `json:"bl" pflag:"true"`
	NestedType       NestedType        `json:"nested"`
	IntArray         []int             `json:"ints" pflag:"[]int{12%2C1}"`
}
	

Available Commands:
              
  help        Help about any command

Flags:
      --bind-default-var     The generated PFlags Set will bind fields to the default variable.
      --default-var string   Points to a variable to use to load default configs. If specified & found, it'll be used instead of the values specified in the tag. (default "defaultConfig")
  -h, --help                 help for pflags
  -p, --package string       Determines the source/destination package. (default ".")

Use "pflags [command] --help" for more information about a command.

2021/11/16 13:22:05 invalid type. it must be struct, received [int] for field [imagePullPolicy]
sandbox_config.go:3: running "pflags": exit status 1
```

**Comment by evalsocket** [comment]
@EngHabu In this implementation user need to define pull policy in integer. Correct ?

**Review by EngHabu — APPROVED** [review]
